com.unity.ml-agents/Runtime/SideChannel/FloatPropertiesChannel.cs

```diff
@@ -42,9 +42,21 @@ public class FloatPropertiesChannel : SideChannel, IFloatProperties
         Dictionary<string, float> m_FloatProperties = new Dictionary<string, float>();
         Dictionary<string, Action<float>> m_RegisteredActions = new Dictionary<string, Action<float>>();
 
-        public override int ChannelType()
+        public FloatPropertiesChannel(int channelId = -1)
         {
-            return (int)SideChannelType.FloatProperties;
+            if (channelId == -1)
+            {
+                ChannelId = (int)ReservedChannelId.FloatProperties;
+            }
+            else
+            {
+                if (channelId < (int)ReservedChannelId.UserSideChannelStart)
```

Comment:
(offline followup). Might be a bit cleaner to make some of these range checks as methods on the enum class. Looks like you can't do it directly, but can do it as an extension: https://docs.microsoft.com/en-us/dotnet/api/system.enum?view=netframework-4.8#Methods

So e.g. `ReservedChannelId.InUserSideChannelRange()`

Reply:
Same for python: https://docs.python.org/3/library/enum.html#allowed-members-and-attributes-of-enumerations